### PR TITLE
Meta: Restore the LibJS Wasm build

### PR DESCRIPTION
**Pull request by alimpfard**
Note that this will fail until we add a "libjs-wasm" branch to github.com/serenityos/libjs-test262-data

### CODE REVIEW

File: .github/workflows/wasm.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      - name: "Show me what you built :)"
+        run: |
+          ls -l ${{ github.workspace }}/Build/wasm
+
+      - name: "Prepare files"
+        run: |
+          cp ${{ github.workspace }}/Build/wasm/bin/libjs.js ${{ github.workspace }}/libjs-test262-data/libjs.js
+          cp ${{ github.workspace }}/Build/wasm/bin/libjs.wasm ${{ github.workspace }}/libjs-test262-data/libjs.wasm
+          echo 'Module.SERENITYOS_COMMIT = "${{ github.sha }}";' >> ${{ github.workspace }}/libjs-test262-data/libjs.js
+          tar -czvf libjs-wasm.tar.gz -C ${{ github.workspace }}/libjs-test262-data .
```
Comment (DanShaders) [inline .github/workflows/wasm.yml:95]: This also adds `${{ github.workspace }}/libjs-test262-data/.git` to the published artifact.

**Comment by DanShaders** [comment]
Two more comments:
- Probably we should store 12Mb bundle in LFS, so we need something similar to [this](https://github.com/SerenityOS/serenity/blob/master/.github/workflows/libjs-test262.yml#L37-L39).
- Looks like Wasm build regressed at some point: https://danshaders.github.io/libjs-test262-data/repl/ (it fails with assertion).

File: .github/workflows/wasm.yml (hunk 1 above)
Comment (alimpfard) [reply .github/workflows/wasm.yml:95]: lol oops

**Comment by DanShaders** [comment]
Turns out "regression" is from improper `js_repl.cpp` fix. This works better: https://github.com/DanShaders/serenity/commit/e330593460001f06ade0c276314187e91e26dfc7#diff-331fa7e034a796854f249809c1b67902ac9f054bc10b4bfd90e65ee75d209d6d (but I have no idea what I've done, I just copied a bunch of lines from command line js repl)

File: .github/workflows/wasm.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        with:
+          python-version: '3.x'
+      - name: "Install Ubuntu dependencies"
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y ninja-build gcc-12 g++-12 libstdc++-12-dev
+      - name: "Install emscripten"
+        uses: mymindstorm/setup-emsdk@v12
+        with:
+          version: 3.1.25
```
Comment (ADKaster) [inline .github/workflows/wasm.yml:27]: Is this the same version we used to have, or latest stable?

File: AK/SipHash.cpp [hunk 3]
```diff
@@ -52,10 +52,11 @@ static void do_siphash(ReadonlyBytes input, u128 key, Bytes output)
     u64 v1 = 0x646f72616e646f6dull;
     u64 v2 = 0x6c7967656e657261ull;
     u64 v3 = 0x7465646279746573ull;
-    auto const left = input.size() & 7;
+    u64 const length = input.size();
+    auto const left = length & 7;
```
Comment (ADKaster) [inline AK/SipHash.cpp:56]: worth splitting the AK 32-bit fix from the js_repl fix?

File: .github/workflows/wasm.yml (hunk 2 above)
Comment (alimpfard) [reply .github/workflows/wasm.yml:27]: it's exactly what we had before the workflow was removed.

File: AK/SipHash.cpp (hunk 3 above)
Comment (alimpfard) [reply AK/SipHash.cpp:56]: Sure, will do.

File: .github/workflows/wasm.yml (hunk 2 above)
Comment (ADKaster) [reply .github/workflows/wasm.yml:27]: Worth an update? or is that for "later :tm: " :)

Comment (alimpfard) [reply .github/workflows/wasm.yml:27]: "later :tm:" sounds fine to me :P

**Review by ADKaster — APPROVED** [review]

**Pull request merged**